### PR TITLE
Remove usage middleware dispatcher in `Route`, `Group` and `MatchingResult`

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ✔️
| Fixed issues  | -

### CODE REVIEW

**Comment by xepozz** [comment]
Already exists in https://github.com/yiisoft/router/pull/194 and https://github.com/yiisoft/router/pull/196

**Comment by vjik** [comment]
> Already exists in #194 and #196

I missed it. May be make it in this PR? It will be easier to add attributes.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 ## 4.0.0 under development
 
+- Chg #210: Remove usage middleware dispatcher in `Route`, `Group` and `MatchingResult` (@vjik)
```
Comment (samdark) [inline CHANGELOG.md:5]:
```suggestion
- Chg #210: Remove usage of middleware dispatcher from `Route`, `Group` and `MatchingResult` (@vjik)
```

**Review by samdark — APPROVED** [review]

**Comment by vjik** [comment]
Closed in favor https://github.com/yiisoft/router/pull/196

**Pull request closed without merging**